java-core/src/main/java/com/pcloud/sdk/internal/RealApiClient.java

```diff
@@ -350,7 +409,27 @@ public Boolean adapt(Response response) throws IOException, ApiError {
             throw new IllegalArgumentException("DownloadOptions parameter cannot be null.");
         }
 
-        Request request = newDownloadLinkRequest(fileId, options);
+        Request request = newDownloadLinkRequest(fileId, null, options);
+
+        return newCall(request, new ResponseAdapter<FileLink>() {
+            @Override
+            public FileLink adapt(Response response) throws IOException, ApiError {
+                return getAsFileLink(response);
+            }
+        });
+
+    }
+
+    @Override
+    public Call<FileLink> createFileLink(String path, DownloadOptions options) {
+        if (path == null) {
```

Comment:
Non-null but empty `path` arguments should also be treated as an illegal argument, so maybe it's a good idea to add a helper function that does the checks and throws with an appropriate message, then replace sanity checks throughout public methods with a `path` argument with the new function.

Reply:
Let's say you want to `listFolder` `/Test/Sub Folder`. There is no trailing slash meaning `listFolder("/Test/Sub Folder/")` results in an error.

To keep this consistent, empty string should reference the root folder since `listFolder("/")` is the trailing slash.

Reply:
If you need to access the content of `/Test/Sub Folder`, why not call the method with the same path without adding the trailing`/`?  Doing a `listfolder` on the parent would also return a `/Test/Sub Folder` for the same child folder.

Reply:
Yeah sure, but I'd say `/` to list the root folder is like `/Test/` to list the Test folder which isn't working.

But yeah,  it's an edge case. If this is needed to get the PR approved I'll be happy to change it. No hard feelings. 😄